x-pack/winlogbeat/module/sysmon/ingest/sysmon.yml

```
@@ -740,15 +740,22 @@ processors:
 
           if (answer.startsWith("type:")) {
             def parts = /\s+/.split(answer);
-            if (parts.length != 3) {
+
+            if (parts.length == 3) {
+              answers.add([
+                "type": params[parts[1]],
+                "data": parts[2]
+              ]);
+              relatedHosts.add(parts[2]);
+            } else if (parts.length == 2) {
+              // Missing data, for example with "type:  33 ;"
+              answers.add([
+                "type": params[parts[1]],
+                "data": ""
+              ]);
+            } else {
               throw new Exception("unexpected QueryResult format");
             }
```

Let's keep the error path up front.
```suggestion
            if (parts.length < 2) {
              throw new Exception("unexpected QueryResult format");
            }
            if (parts.length == 3) {
              answers.add([
                "type": params[parts[1]],
                "data": parts[2]
              ]);
              relatedHosts.add(parts[2]);
            } else {
              answers.add([
                "type": params[parts[1]],
                "data": ""
              ]);
            }
```
I'm not sure that we want to add the empty `data` value. I can see some reasons for it, but they are not strong.

---

I wasn't sure about the empty `data` either, it's now removed.